### ISSUE TITLE
Issue with pretrained models and CNTK backend

### ISSUE THREAD

**Issue by miguelgfierro** [fchollet/keras#7148]
With the last version of keras (the master of today) and CNTK v2.0 with GPU 1SGD, I get an error when:
```python
model = ResNet50(include_top=False)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-70030c8bcad1> in <module>()
----> 1 model = ResNet50(include_top=False)#doesn't work with CNTK backend

/anaconda/envs/strata/lib/python3.5/site-packages/keras/applications/resnet50.py in ResNet50(include_top, weights, input_tensor, input_shape, pooling, classes)
    204         bn_axis = 1
    205 
--> 206     x = ZeroPadding2D((3, 3))(img_input)
    207     x = Conv2D(64, (7, 7), strides=(2, 2), name='conv1')(x)
    208     x = BatchNormalization(axis=bn_axis, name='bn_conv1')(x)

/anaconda/envs/strata/lib/python3.5/site-packages/keras/engine/topology.py in __call__(self, inputs, **kwargs)
    594 
    595             # Actually call the layer, collecting output(s), mask(s), and shape(s).
--> 596             output = self.call(inputs, **kwargs)
    597             output_mask = self.compute_mask(inputs, previous_mask)
    598 

/anaconda/envs/strata/lib/python3.5/site-packages/keras/layers/convolutional.py in call(self, inputs)
   1354         return K.spatial_2d_padding(inputs,
   1355                                     padding=self.padding,
-> 1356                                     data_format=self.data_format)
   1357 
   1358     def get_config(self):

/anaconda/envs/strata/lib/python3.5/site-packages/keras/backend/cntk_backend.py in spatial_2d_padding(x, padding, data_format)
   1713         if num_dynamic_axis > 0:
   1714             assert len(base_shape) == 3
-> 1715             x = _padding(x, padding[0], 0)
   1716             x = _padding(x, padding[1], 1)
   1717         else:

/anaconda/envs/strata/lib/python3.5/site-packages/keras/backend/cntk_backend.py in _padding(x, pattern, axis)
   1673                          'shape `%s` contains non-specified dimension, '
   1674                          'which is not supported. Please give fixed '
-> 1675                          'dimension to enable padding.' % base_shape)
   1676     if pattern[0] > 0:
   1677         prefix_shape = list(base_shape)

TypeError: not all arguments converted during string formatting


```
I tried with TF backend and it works correctly

**Comment by souptc** [comment]
The reason is cntk doesn't support padding with non-specified input shape (which is the default option in ResNet50), so you need to specify the input shape, please see the 'test_resnet50_notop_specified_input_shape' test case in application_test.py as example.

We will support this functionality with None shape in next release.

The reason of this error is because I made a stupid mistake when create the error message. I will fix this part.

**Comment by fboylu** [comment]
Hi, 
We tried this again with cntk 2.1 and still getting an error:

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-17-3bf2f8edb0e6> in <module>()
----> 1 model = ResNet50(include_top=False)

/anaconda/envs/py35/lib/python3.5/site-packages/keras/applications/resnet50.py in ResNet50(include_top, weights, input_tensor, input_shape, pooling, classes)
    208     x = BatchNormalization(axis=bn_axis, name='bn_conv1')(x)
    209     x = Activation('relu')(x)
--> 210     x = MaxPooling2D((3, 3), strides=(2, 2))(x)
    211 
    212     x = conv_block(x, 3, [64, 64, 256], stage=2, block='a', strides=(1, 1))

/anaconda/envs/py35/lib/python3.5/site-packages/keras/engine/topology.py in __call__(self, inputs, **kwargs)
    600 
    601             # Actually call the layer, collecting output(s), mask(s), and shape(s).
--> 602             output = self.call(inputs, **kwargs)
    603             output_mask = self.compute_mask(inputs, previous_mask)
    604 

/anaconda/envs/py35/lib/python3.5/site-packages/keras/layers/pooling.py in call(self, inputs)
    152                                         strides=self.strides,
    153                                         padding=self.padding,
--> 154                                         data_format=self.data_format)
    155         return output
    156 

/anaconda/envs/py35/lib/python3.5/site-packages/keras/layers/pooling.py in _pooling_function(self, inputs, pool_size, strides, padding, data_format)
    215         output = K.pool2d(inputs, pool_size, strides,
    216                           padding, data_format,
--> 217                           pool_mode='max')
    218         return output
    219 

/anaconda/envs/py35/lib/python3.5/site-packages/keras/backend/cntk_backend.py in pool2d(x, pool_size, strides, padding, data_format, pool_mode)
   1567             pool_size,
   1568             strides,
-> 1569             auto_padding=[padding])
   1570     elif pool_mode == 'avg':
   1571         x = C.pooling(

/anaconda/envs/py35/lib/python3.5/site-packages/cntk/internal/swig_helper.py in wrapper(*args, **kwds)
     67     @wraps(f)
     68     def wrapper(*args, **kwds):
---> 69         result = f(*args, **kwds)
     70         map_if_possible(result)
     71         return result

/anaconda/envs/py35/lib/python3.5/site-packages/cntk/ops/__init__.py in pooling(operand, pooling_type, pooling_window_shape, strides, auto_padding, ceil_out_dim, include_pad, name)
    414     pooling_window_shape, strides, auto_padding = sanitize_pooling_args(pooling_window_shape, strides, auto_padding)
    415     return pooling(operand, pooling_type, pooling_window_shape, strides, auto_padding,
--> 416                    ceil_out_dim, include_pad, name)
    417 
    418 

ValueError: Convolution operation requires that kernel dim 3 <= input dim 1.

[CALL STACK]
[0x7efdaac972ac]                                                       + 0x5762ac
[0x7efdaadd193c]    Microsoft::MSR::CNTK::ConvolveGeometry::  ComputeOutputShape  (Microsoft::MSR::CNTK::TensorShape const&,  Microsoft::MSR::CNTK::TensorShape const&,  Microsoft::MSR::CNTK::TensorShape const&,  Microsoft::MSR::CNTK::TensorShape const&,  std::vector<bool,std::allocator<bool>> const&,  std::vector<bool,std::allocator<bool>> const&,  Microsoft::MSR::CNTK::TensorShape const&,  Microsoft::MSR::CNTK::TensorShape const&,  bool) + 0x5bc
[0x7efdaadc14e6]    CNTK::PrimitiveFunction::  ConvolutionOpOutputShape  (CNTK::PrimitiveOpType,  CNTK::NDShape const&,  CNTK::NDShape&,  CNTK::NDShape&,  CNTK::NDShape&,  std::vector<bool,std::allocator<bool>>&,  std::vector<bool,std::allocator<bool>>&,  CNTK::NDShape&,  CNTK::NDShape&,  bool,  bool,  bool) + 0x136
[0x7efdaadc887b]    CNTK::PrimitiveFunction::  InferOutputs  (std::vector<CNTK::Variable,std::allocator<CNTK::Variable>>&) + 0x287b
[0x7efdaad9bf6f]                                                       + 0x67af6f
[0x7efe0644ca99]                                                       + 0xea99
[0x7efdaad96f9e]    CNTK::Function::  InitOutputs  ()                  + 0x5e
[0x7efdaae01731]    CNTK::CompositeFunction::  InferOutputs  (std::vector<CNTK::Variable,std::allocator<CNTK::Variable>>&) + 0x11
[0x7efdaad9bf6f]                                                       + 0x67af6f
[0x7efe0644ca99]                                                       + 0xea99
[0x7efdaad96f9e]    CNTK::Function::  InitOutputs  ()                  + 0x5e
[0x7efdaad9c748]    CNTK::  AsComposite  (std::shared_ptr<CNTK::Function> const&,  std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>> const&) + 0x2a8
[0x7efdaad9feb4]    CNTK::  UnaryOp  (CNTK::PrimitiveOpType,  CNTK::Variable const&,  CNTK::Dictionary&&,  std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>> const&) + 0x144
[0x7efdaada1f81]    CNTK::  Pooling  (CNTK::Variable const&,  CNTK::PoolingType,  CNTK::NDShape const&,  CNTK::NDShape const&,  std::vector<bool,std::allocator<bool>> const&,  bool,  bool,  std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>> const&) + 0x341
[0x7efdab886622]                                                       + 0x251622
[0x7efe067175e9]    PyCFunction_Call                                   + 0xf9
[0x7efe0679ebd5]    PyEval_EvalFrameEx                                 + 0x8fb5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe066f5661]                                                       + 0x9a661
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe0679c234]    PyEval_EvalFrameEx                                 + 0x6614
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe066f5661]                                                       + 0x9a661
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe0679c234]    PyEval_EvalFrameEx                                 + 0x6614
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe066f5542]                                                       + 0x9a542
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe066de77c]                                                       + 0x8377c
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe06732e3f]                                                       + 0xd7e3f
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe0679c314]    PyEval_EvalFrameEx                                 + 0x66f4
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe0679fd1b]    PyEval_EvalCode                                    + 0x3b
[0x7efe06792dfe]                                                       + 0x137dfe
[0x7efe067175e9]    PyCFunction_Call                                   + 0xf9
[0x7efe0679ebd5]    PyEval_EvalFrameEx                                 + 0x8fb5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe066f5661]                                                       + 0x9a661
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe0679c234]    PyEval_EvalFrameEx                                 + 0x6614
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679f166]    PyEval_EvalFrameEx                                 + 0x9546 (x2)
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe066f5661]                                                       + 0x9a661
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe0679c234]    PyEval_EvalFrameEx                                 + 0x6614
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe066f5661]                                                       + 0x9a661
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe0679c234]    PyEval_EvalFrameEx                                 + 0x6614
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679f166]    PyEval_EvalFrameEx                                 + 0x9546
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe066f5661]                                                       + 0x9a661
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe0679c234]    PyEval_EvalFrameEx                                 + 0x6614
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679f166]    PyEval_EvalFrameEx                                 + 0x9546
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679f166]    PyEval_EvalFrameEx                                 + 0x9546
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe0679fd1b]    PyEval_EvalCode                                    + 0x3b
[0x7efe06792dfe]                                                       + 0x137dfe
[0x7efe067175e9]    PyCFunction_Call                                   + 0xf9
[0x7efe0679ebd5]    PyEval_EvalFrameEx                                 + 0x8fb5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679edf5]    PyEval_EvalFrameEx                                 + 0x91d5
[0x7efe0679fb49]                                                       + 0x144b49
[0x7efe0679fcd8]    PyEval_EvalCodeEx                                  + 0x48
[0x7efe066f5542]                                                       + 0x9a542
[0x7efe066c2236]    PyObject_Call                                      + 0x56
[0x7efe067e08a2]                                                       + 0x1858a2
[0x7efe067e1565]    Py_Main                                            + 0x945
[0x400add]          main                                               + 0x15d
[0x7efe0577c830]    __libc_start_main                                  + 0xf0
[0x4008b9]

**Comment by souptc** [comment]
hi, this issue is fixed with cntk v2.2 release, please see pr #7907 
Could you update to cntk v2.2 and have a try?

**Comment by fboylu** [comment]
Thanks, worked with the parameter input_shape=(224,224,3).
